runtime/tests/checker/interface_test.go

```diff
@@ -1525,317 +1525,132 @@ func TestCheckInterfaceSelfUse(t *testing.T) {
-              }
+              enum Nested: Int {}
           }
         `,
-	)
-
-	errs := RequireCheckerErrors(t, err, 1)
-
-	assert.IsType(t, &sema.ConformanceError{}, errs[0])
-}
+		)
 
-func TestCheckInvalidContractInterfaceTypeRequirementConformanceMissingFunction(t *testing.T) {
+		errs := RequireCheckerErrors(t, err, 1)
 
-	t.Parallel()
+		assert.IsType(t, &sema.InvalidNestedDeclarationError{}, errs[0])
+	})
 
-	_, err := ParseAndCheck(t,
-		`
+	t.Run("contract", func(t *testing.T) {
+		_, err := ParseAndCheck(t,
+			`
           contract interface Test {
-
-              struct interface NestedInterface {
-                  fun test(): Bool
-              }
-
-              struct Nested: NestedInterface {}
-          }
-
-          contract TestImpl: Test {
-
-              struct Nested: Test.NestedInterface {
-                  // missing function 'test'
-              }
+              contract Nested {}
           }
         `,
-	)
-
-	errs := RequireCheckerErrors(t, err, 1)
-
-	assert.IsType(t, &sema.ConformanceError{}, errs[0])
-}
+		)
 
-func TestCheckInvalidContractInterfaceTypeRequirementMissingConformance(t *testing.T) {
+		errs := RequireCheckerErrors(t, err, 1)
 
-	t.Parallel()
+		assert.IsType(t, &sema.InvalidNestedDeclarationError{}, errs[0])
+	})
 
-	_, err := ParseAndCheck(t,
-		`
+	t.Run("contract interface", func(t *testing.T) {
+		_, err := ParseAndCheck(t,
+			`
           contract interface Test {
-
-              struct interface NestedInterface {
-                  fun test(): Bool
-              }
-
-              struct Nested: NestedInterface {}
-          }
-
-          contract TestImpl: Test {
-
-              // missing conformance to 'Test.NestedInterface'
-              struct Nested {
-                  fun test(): Bool {
-                      return true
-                  }
-              }
+              contract interface Nested {}
           }
         `,
-	)
-
-	errs := RequireCheckerErrors(t, err, 1)
-
-	assert.IsType(t, &sema.MissingConformanceError{}, errs[0])
-}
-
-func TestCheckContractInterfaceTypeRequirementImplementation(t *testing.T) {
-
-	t.Parallel()
-
-	_, err := ParseAndCheck(t,
-		`
-          struct interface OtherInterface {}
-
-          contract interface Test {
-
-              struct interface NestedInterface {
-                  fun test(): Bool
-              }
-
-              struct Nested: NestedInterface {}
-          }
-
-          contract TestImpl: Test {
+		)
 
-              struct Nested: Test.NestedInterface, OtherInterface {
-                  fun test(): Bool {
-                      return true
-                  }
-              }
-          }
-        `,
-	)
+		errs := RequireCheckerErrors(t, err, 1)
 
-	require.NoError(t, err)
+		assert.IsType(t, &sema.InvalidNestedDeclarationError{}, errs[0])
+	})
 }
 
-func TestCheckContractInterfaceFungibleToken(t *testing.T) {
+// TODO: re-enable this test with the v2 fungible token contract
```

Comment:
Might be good to create an issue for enabling these tests, and add it to the stable-cadence epic, so that we don't forget.

Reply:
https://github.com/onflow/cadence/issues/2719